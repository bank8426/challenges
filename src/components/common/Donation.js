import React from 'react'
import styled from 'styled-components';
import PropTypes from 'prop-types'

const StyledDonation = styled.h3`
  margin: 1em;
`;

const Donation = ({donate}) => {
  return (
    <StyledDonation>
      All donations: {donate.toLocaleString()} THB
    </StyledDonation>
  )
}

Donation.defaultProps = {
  donate: 0,
}

Donation.propTypes={
  donate : PropTypes.number.isRequired,
}

export default Donation
