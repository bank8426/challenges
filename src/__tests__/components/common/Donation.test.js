import React from 'react'
import Donation from '../../../components/common/Donation'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

it('should display total 0 THB', () => {
  const tree = renderer.create(<Donation />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should display total 100 THB', () => {
  const tree = renderer.create(<Donation donate={100}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

