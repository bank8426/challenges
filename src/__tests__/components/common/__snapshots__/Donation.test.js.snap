// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display total 0 THB 1`] = `
.c0 {
  margin: 1em;
}

<h3
  className="c0"
>
  All donations: 
  0
   THB
</h3>
`;

exports[`should display total 100 THB 1`] = `
.c0 {
  margin: 1em;
}

<h3
  className="c0"
>
  All donations: 
  100
   THB
</h3>
`;
